code/modules/mob/living/simple_mob/subtypes/animal/space/carp.dm

```diff
@@ -0,0 +1,125 @@
+// Space carp show up as a random event to wreck hapless people in space or near windows.
+// They generally fit the archetype of 'fast but fragile'.
+// This is compensated by being in groups (usually).
+/mob/living/simple_mob/animal/space/carp
+	name = "space carp"
+	desc = "A ferocious, fang-bearing creature that resembles a fish."
+	icon_state = "carp"
+	icon_living = "carp"
+	icon_dead = "carp_dead"
+	icon_gib = "carp_gib"
+
+	faction = "carp"
+	maxHealth = 25
+	health = 25
+	movement_cooldown = 0 // Carp go fast
+	hovering = TRUE.
+
+	response_help = "pets the"
+	response_disarm = "gently pushes aside the"
+	response_harm = "hits the"
+
+	melee_damage_lower = 7 // About 14 DPS.
+	melee_damage_upper = 7
+	base_attack_cooldown = 10 // One attack a second.
+	attack_sharp = TRUE
+	attack_sound = 'sound/weapons/bite.ogg'
+	attacktext = list("bitten")
+
+	meat_amount = 1
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/carpmeat
+
+	ai_holder_type = /datum/ai_holder/simple_mob/melee
+
+	var/knockdown_chance = 15
+
+/mob/living/simple_mob/animal/space/carp/Process_Spacemove(var/check_drift = 0)
+	return TRUE //No drifting in space for space carp!	//original comments do not steal	// stolen.
+
+/mob/living/simple_mob/animal/space/carp/apply_melee_effects(var/atom/A)
+	if(isliving(A))
+		var/mob/living/L = A
+		if(prob(knockdown_chance))
+			L.Weaken(3)
+			L.visible_message(span("danger", "\The [src] knocks down \the [L]!"))
+
+
+// Subtypes.
+/mob/living/simple_mob/animal/space/carp/large
+	name = "elder carp"
+	desc = "An older, more matured carp. Few survive to this age due to their aggressiveness."
+	icon = 'icons/mob/64x32.dmi'
+	icon_state = "shark"
+	icon_living = "shark"
+	icon_dead = "shark_dead"
+
+	maxHealth = 50
+	health = 50
+	movement_cooldown = 10 // Slower than the younger carp.

```

Comment:
10 movement cooldown means it waits about one second to step.
It can be moved to 5 per tile, but otherwise it looked weird to me if the fatter carp moved the same speed as their smaller counterparts.

Reply:
That's a fair point, but I'd argue that it looks enough like a miniboss to warrant being treated like one.

Reply:
Would 5 movement cooldown on this carp and 10 on the /huge type be better?

Reply:
Sounds alright, yeah.